packages/api-server/src/routes/note.ts

```diff
@@ -52,17 +52,38 @@ router.get(
   "/bulkGet",
   asyncHandler(async (req: Request, res: Response<BulkGetNoteResp>) => {
     const { ids, ws } = req.query as unknown as EngineBulkGetNoteRequest;
+
+    // Express will convert the ids array into an object if it exceeds the
+    // 'arrayLimit' value set in the query parser (see {@link appModule}). In
+    // that case, convert the object back to an array
+    let processedIdPayload = ids;
+    if (!Array.isArray(processedIdPayload)) {
+      processedIdPayload = Object.values(processedIdPayload);
+    }
```

Comment:
Maybe extract a local function so that all occurences point to the comment explanation.
```suggestion
    // Express will convert the ids array into an object if it exceeds the
    // 'arrayLimit' value set in the query parser (see {@link appModule}). In
    // that case, convert the object back to an array
    // See: https://github.com/ljharb/qs#parsing-arrays, https://github.com/expressjs/express/issues/2661
    function toArray(ids: Record<string, string> | string[]) {
      if (!Array.isArray(ids)) {
        return Object.values(ids);
      }
      return ids
    }
```